### PR TITLE
[Fix] Credentials Encryption

### PR DESCRIPTION
**Pull request by gustavohc07**
This PR was made to solve issues when deploying PredictionBook to staging and production:
- Adds production and staging encrypted credentials
- Does not use credentials for development purposes, i.e. development and test environment

### CODE REVIEW

**Comment by gustavohc07** [comment]
> Is this so we can remove our private predictionbook deployment repo?

The main reason is the way Rails 6 handles credentials that changes from Rails 5 and it was breaking deployment. I'm not sure with only the credentials encrypted we are able to remove our private deployment repo. Because we got some other configurations regarding deployment.

Although, we are able to pack all the information we need to build the deployment files inside an encrypted file 🤔. That way we might be able to remove the private repository.

**Comment by michaelwebb76** [comment]
I think the only other thing in the deployment repo is the capistrano setup. @cattruscott is there anything sensitive in that config that we couldn't just port into this repo?

**Review by michaelwebb76 — APPROVED** [review]

File: config/application.rb [hunk 1]
```diff
@@ -56,8 +60,10 @@ class Application < Rails::Application
       end
     end
 
-    config.before_initialize do
-      Credentials.load_config
+    if Rails.env.development? || Rails.env.test?
```
Comment (cdaloisio) [inline config/application.rb:63]: If we can avoid addition this conditional, that would be better. Ideally, if we want people to create their own development credentials, we can do that by providing a sample file, that with a rake task will generate the development key and the development encrypted credentials. Then all the custom environment wrangling can be removed.

**Review by cdaloisio — APPROVED** [review]
Pre-approving. Just the change that we chatted about in our catch up this morning and then ship it!

**Pull request closed without merging**